Tensile/SolutionStructs.py

```diff
@@ -2835,15 +2868,46 @@ def assignDerivedParameters(state):
     if "MatrixInstK" in state:
       state["LoopIters"] //= state["MatrixInstK"]
 
-    # LoopIters should greater than PrefetchLocalRead
-    if (state["LoopIters"] - state["PrefetchLocalRead"]) < 1:
-      reject(state, "LoopIters %u should greater than PrefetchLocalRead %u" \
-        % (state["LoopIters"],state["PrefetchLocalRead"]))
+    if state["LoopIters"] < 1:
+      reject(state, "LoopIters need to greater than 0")
+      return
+
+    # PLR > 2xLoopIters is redundant setting
+    if state["PrefetchLocalRead"] >= 2*state["LoopIters"]:
+      reject(state, "PrefetchLocalRead %u larger than 2x LoopIters %u" % (state["PrefetchLocalRead"],state["LoopIters"]))
+
+    # prefetch wider read iteration > LoopIters, no enough iterations for prefetching
+    if state["EnableMatrixInstruction"]:
+      if (state["PrefetchLocalRead"]%state["LoopIters"])*state["LocalReadVectorWidth"]//state["ProblemType"]["DataType"].numMIInput() >= state["LoopIters"]:
+        reject(state, "with PrefetchLocalRead %u LoopIters %u LocalReadVectorWidth %u, not enough LoopIters to prefetch %ux%u iterations, " \
+          % (state["PrefetchLocalRead"],state["LoopIters"],state["LocalReadVectorWidth"],\
+          (state["PrefetchLocalRead"]%state["LoopIters"]),state["LocalReadVectorWidth"]//state["ProblemType"]["DataType"].numMIInput()))
 
     # reject conditions with lower performance
     if state["ScheduleIterAlg"] == 2 and \
     (state["ExpandPointerSwap"] != 1 or state["LoopIters"] != 1 or state["ScheduleGlobalRead"] != 1):
-      reject(state, "ScheduleIterAlg 2 only work with EPS1_SGW1, LoopIter=1")
+      reject(state, "ScheduleIterAlg 2 only work with EPS1_SGR1, LoopIter=1")
+
+    if state["TransposeLDS"] == 1:
+      if not state["MatrixInstruction"]:
```

Comment:
I think you meant EnableMatrixInstruction here?

Reply:
changed